demos/smart_classroom_demo/src/action_detector.cpp

```diff
@@ -131,23 +131,32 @@ DetectedActions ActionDetection::fetchResults() {
     const auto loc_blob_name = new_network_ ? config_.new_loc_blob_name : config_.old_loc_blob_name;
     const auto det_conf_blob_name = new_network_ ? config_.new_det_conf_blob_name : config_.old_det_conf_blob_name;
 
-    const cv::Mat priorbox_out =
-        new_network_
-          ? cv::Mat()
-          : cv::Mat(ieSizeToVector(request->GetBlob(config_.old_priorbox_blob_name)->getTensorDesc().getDims()),
-                    CV_32F, request->GetBlob(config_.old_priorbox_blob_name)->buffer());
+    cv::Mat priorboxMat;
+    if (new_network_) {
+        priorboxMat = cv::Mat();
+    } else {
+        LockedMemory<void> priorboxOutBlobMapped = as<MemoryBlob>(request->
+                                                                  GetBlob(config_.old_priorbox_blob_name))->rwmap();
+        priorboxMat = cv::Mat(ieSizeToVector(request->
+                              GetBlob(config_.old_priorbox_blob_name)->getTensorDesc().getDims()), CV_32F,
+                              priorboxOutBlobMapped);
```

Comment:
I don't think this is correct. `priorboxMat` is initialized with a pointer to the blob data, which becomes invalid once the `else` block ends.